string[] names
float64[] positions #order: shoulder_pan, elbow, upper_arm_roll, shoulder_lift
float64[] margins
float64 timeout